src/config/configure.in

```diff
@@ -1738,16 +1738,16 @@ then
    then
       if test "$hypre_using_openmp" = "yes"
       then
-         AC_CHECK_PROGS(CXX, [xlC_r xlc_r xlC xlc icpc icc icpx icx g++ gcc pgCC pgcc pgc++ CC cc KCC kcc])
+         AC_CHECK_PROGS(CXX, [xlC_r xlc_r xlC xlc icx icpc icc icpx g++ gcc pgCC pgcc pgc++ CC cc KCC kcc])
```

Comment:
Even though icc is gone, icx can be picked  from the list. Not sure why we need to reorder.

Reply:
Some systems have a residual `icc` installation lying around. If we test for icx first, the residual `icc` wouldn't be a problem.

Reply:
Again, they can always do CC=WHATEVER, and after icc is complete gone, we just delete it from the list.

Reply:
Hi @victorapm I don't really have that strong opinion on this PR :) My main concern if it is worth our time to change and test for some thing that is working OK. I am a "minimalist" :). Thanks!

Reply:
If we are reordering 'icc' (paired with 'icpc') and 'icx' (paired with 'icpx'), why are we not also reordering 'mpiicc' and 'mpiicx'?  I am not opposed to switching the order of these if 'icc' is going away, but shouldn't we do this consistently?  We tried to keep the mpi versions consistent with the raw compilers before.

Reply:
Good catch. `mpiicc` and `mpiicx` should be reordered as well. I can do that if we decide this PR is useful

Reply:
```
icpx icx icpc icc
mpiicx mpiicc
```
looks good to me.

Reply:
Done in https://github.com/hypre-space/hypre/pull/1024/commits/91583f12640eeb455c152e153d632f4c58458d48